ArduCopter/mode_guided.cpp

```diff
@@ -384,6 +396,28 @@ void ModeGuided::run()
     }
  }
 
+// guided_pretakeoff_run - pretakeoff in guided mode
+//      called by guided_run at 100hz or more
+void ModeGuided::pretakeoff_run()
+{
```

Comment:
```suggestion
{    // if not armed set throttle to zero and exit immediately
    if (!motors->armed() || !copter.ap.auto_armed) {
        make_safe_spool_down();
        return;
    }
```
We should add this check. Without the check there is undesirable behavior if the user disarms in pretakeoff state. Reproducible in sitl: takeoff command -> disarm while spooling -> rearm and watch the heli spool up without a takeoff command